Style Guide and Diataxis Review: Streams IV (#226)
iotaledger/streams: documentation/docs/libraries/c/getting_started.md

- C
- cmake
- std
- reference
---
# Getting Started
Using the C bindings, you can build a Streams API that is pulled into other languages. The streams instance used for the bindings is built with the `sync-client` flag to ensure a compatible client interface using the `iota.rs iota-client` crate. 

## Prerequisites

Before building anything, make sure you have [`cmake`]((https://cmake.org/)) installed on your machine.

## Build the Library

First, to build the library, make sure you are in the c directory:

```
cd bindings/c
``` 
Then update the flags in the `CMakeLists.txt` and run ```cmake .``` to prepare the installation files. 

### Options for CMakeLlists.txt

- `NO_STD`: Enable no_std build, without iota_client. When ON, `SYNC_CLIENT` is not supported.
- `SYNC_CLIENT`: Enable sync transport via iota_client, otherwise it is going to be Bucket which can only be used for tests.
- `STATIC`: Build static library when ON, otherwise dynamic library.
- `RELEASE`: Build in release or debug mode. When ON, builds release; when OFF, build debug.

To build the library, you run:

```bash 
make
```

> This generates a binary library for inclusion into a project. This can be either: 

Author: 👍

- `iota_streams_c_static`
- `iota_streams_c.so` (Unix)
- `iota_streams_c.dll` (Windows)

An example of the header file can be found in `include/channels.h`.

## Starting a Channel 

> Once the package has been built, you can pull it into a script file: 

Reviewer: ```suggestion
Once you have [built](#build-the-library) the package, you can pull it into a script file:
```
